framework/sandstone_tests.cpp

```diff
@@ -225,7 +227,26 @@ SandstoneTestSet::TestSet SandstoneTestSet::add_test_list(const char *fname, std
     std::vector<struct test_cfg_info> entries = load_test_list(list_file, this, cfg.ignore_unknown_tests, errors);
     if (!errors.empty()) return res;
     for (auto e : entries) {
-        test_set.push_back(e.test);
+        if (!e.attribute.empty()
+                && !std::ranges::equal(e.attribute, std::string("default"), [] (const char& a, const char& b) { return (std::tolower(a) == std::tolower(b)); })) {
```

Comment:
Line too long, please split.

Please don't use `std::ranges::equal` with a char-by-char lowercasing. That's rule 1 of lowercasing: it's not a character operation. Instead, just do a `strcasecmp`.

`std::ranges::equal` was added in GCC 11. I think all our internal targets support that, but I'm not sure.